salt/minion.py

```diff
@@ -198,6 +199,20 @@ def resolve_dns(opts, fallback=True):
     return ret
 
 
+def srv_lookup(name, protocol='TCP', domain=None):
+    if domain:
+        domain = '.' + domain
+    record_name = '_{0}_._{1}{2}'.format(name, protocol.lower(), domain or '')
+    cmd = ['dig', '+short', record_name, 'SRV']
+    try:
+        result = subprocess.check_output(cmd)
+    except subprocess.CalledProcessError as err:
+        raise err
+    else:
+        regex = re.compile('(?P<prio>\d+) (?P<weight>\d+) (?P<port>\d+) (?P<host>.+)')
+        return [match.groupdict() for match in regex.finditer(result)]
```

Comment:
If we care about ordering in the record itself, then something along the lines of:

```
records = [match.groupdict() for match in regex.finditer(result)]
return sorted(records, key=lambda k: (int(k['prio']), -int(k['weight'])))

```

This would return records based on priority (lowest preferred) and then weight (higher preferred) .